### PR TITLE
[fei4960.4.examples] Migrate examples

### PR DESCRIPTION
**Pull request by somewhatabstract**
## Summary:
This migrates the various test examples from Khan/render-gateway and adds the infrastructure to run them (i.e. `babel-watch`).

Issue: FEI-4960

## Test plan:
`yarn typecheck`
`yarn build`
`yarn build:types`
`yarn --cwd packages/wonder-stuff-render-environment-jsdom example jsdom-simple`
`yarn --cwd packages/wonder-stuff-render-server example simple`
`yarn --cwd packages/wonder-stuff-render-server example error`
`yarn --cwd packages/wonder-stuff-render-server example logging`

### CODE REVIEW

**Comment by somewhatabstract** [comment]
The parent pull-request ([#608](https://github.com/Khan/wonder-stuff/pull/608)) has been merged into `main`, but this branch (`fei4960.4.examples`) now has conflicts with the new base branch. These conflicts must be resolved before checks can complete on this pull-request.

File: packages/wonder-stuff-core/tsconfig.json [hunk 1]
```diff
@@ -4,5 +4,7 @@
   "compilerOptions": {
     "outDir": "dist",
     "rootDir": "src",
-  }
+  },
+  "references": [
+  ]
```
Comment (somewhatabstract) [inline packages/wonder-stuff-core/tsconfig.json:9]: **note:** Remnants from trying to fix a small nit. Seemed better to leave it for now as it makes adding references easier in the long term and its inert.

File: packages/wonder-stuff-i18n/tsconfig.json [hunk 2]
```diff
@@ -4,5 +4,7 @@
   "compilerOptions": {
     "outDir": "dist",
     "rootDir": "src",
-  }
+  },
+  "references": [
+  ]
```
Comment (somewhatabstract) [inline packages/wonder-stuff-i18n/tsconfig.json:9]: **note:** Remnants from trying to fix a small nit. Seemed better to leave it for now as it makes adding references easier in the long term and its inert.

File: packages/wonder-stuff-render-environment-jsdom/examples/README.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Examples
+
+Each folder in this location must contain a file named `run.js` that is
```
Comment (somewhatabstract) [inline packages/wonder-stuff-render-environment-jsdom/examples/README.md:3]: **nit:**
```suggestion
Each folder in this location must contain a file named `run.ts` that is
```

File: packages/wonder-stuff-render-environment-jsdom/examples/README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+responsible for executing the given example. To execute an example, you can
+run the following command from the root folder, where `<EXAMPLE>` is the name
+of the example folder.
+
+```shell
+yarn start:example <EXAMPLE>
```
Comment (somewhatabstract) [inline packages/wonder-stuff-render-environment-jsdom/examples/README.md:9]: **nit:** The command is currently `yarn example` for brevity of execution.
```suggestion
yarn example <EXAMPLE>
```

File: packages/wonder-stuff-render-server/examples/README.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# Examples
+
+Each folder in this location must contain a file named `run.js` that is
```
Comment (somewhatabstract) [inline packages/wonder-stuff-render-server/examples/README.md:3]: **nit:** 
```suggestion
Each folder in this location must contain a file named `run.ts` that is
```

File: packages/wonder-stuff-render-server/examples/README.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+responsible for executing the given example. To execute an example, you can
+run the following command from the root folder, where `<EXAMPLE>` is the name
+of the example folder.
+
+```shell
+yarn start:example <EXAMPLE>
```
Comment (somewhatabstract) [inline packages/wonder-stuff-render-server/examples/README.md:9]: nit: The command is currently yarn example for brevity of execution.
```suggestion
yarn example <EXAMPLE>
```

**Review by jeresig — APPROVED** [review]

**Pull request merged**